### PR TITLE
Story3 t2

### PR DESCRIPTION
**Pull request by jennymegan**
Profile page set up, with logic

### CODE REVIEW

File: src/components/profile/profile.css [hunk 1]
```diff
@@ -1,7 +1,167 @@
...
+}
+
+.form_row button {
+    background: gray;
+    color: white;
+    border: 0;
+    width: 70%;
+}
+
+
```
Comment (elstevens23) [inline src/components/profile/profile.css]: bcus im annoying and being pedantic pls remove this extra line break :)

File: src/components/profile/profile.css [hunk 2]
```diff
@@ -1,7 +1,167 @@
...
\ No newline at end of file
+    z-index: 20;
+    display: flex;
+    justify-content: start;
+    align-items: center;
+    text-align: center;
+}
+
+
+
```
Comment (elstevens23) [inline src/components/profile/profile.css]: oof extra line breaks 👎

File: src/components/profile/profile.js [hunk 3]
```diff
@@ -6,12 +6,27 @@ import {Branding} from "../branding";
 import {UserInfo} from "../userInfo";
 import axios from "axios";
 import './profile.css';
+import Header from "../header";
+import TextareaInput from "../textareaInput";
+import Button from "../button";
+
```
Comment (elstevens23) [inline src/components/profile/profile.js]: are these some extra unnecessary line breaks i see

File: src/components/profile/profile.js [hunk 4]
```diff
@@ -6,12 +6,27 @@ import {Branding} from "../branding";
...
     constructor(props) {
         super(props);
-        this.state = {url: props.match.params.user, loggedInUser: false, UserProfile: {username: '', bio: ''}, loggedInUserProfile: false, responsedata: false};
+        this.state = {url: props.match.params.user,
+                    loggedInUser: false,
+                    tilPost: '',
+                    userTils: [],
+                    UserProfile: {
+                        username: '',
+                        bio: "I'm a happy little vegemite"
```
Comment (elstevens23) [inline src/components/profile/profile.js]: i love this but DELETE

File: src/components/profile/profile.js [hunk 5]
```diff
@@ -35,7 +50,7 @@ class Profile extends Component {
             if (response.data.success) {
                 this.setState({UserProfile: response.data.info})
             } else {
-                this.setState({UserProfile: {username: 'error could not find user', bio: ''}})
+                this.setState({UserProfile: {username: 'error could not find user', bio: "I'm a happy little vegemite"}})
```
Comment (elstevens23) [inline src/components/profile/profile.js]: noooooooooooooooooooo

File: src/components/profile/profile.js [hunk 6]
```diff
@@ -64,15 +80,125 @@ class Profile extends Component {
...
+
+    formatDate(date) {
+       const splitTimestamp = date.split(/[T.]/);
+       const dateOfTil = splitTimestamp[0];
+       let timeOfTil = splitTimestamp[1];
+       timeOfTil = timeOfTil.substring(0, timeOfTil.length-3)
+       return dateOfTil + ' at ' + timeOfTil
+    }
+
+
```
Comment (elstevens23) [inline src/components/profile/profile.js]: honestly wtf dis line break

File: src/components/profile/profile.js [hunk 7]
```diff
@@ -64,15 +80,125 @@ class Profile extends Component {
...
+    displayTILs() {
+        const userTils = this.state.userTils;
+        userTils.reverse()
+       return (
```
Comment (elstevens23) [inline src/components/profile/profile.js]: woah this uneven tabbing makes me feel nauseous

File: src/components/profile/profile.js [hunk 8]
```diff
@@ -64,15 +80,125 @@ class Profile extends Component {
...
+                          <p>Posted at: {this.formatDate(post.createdAt)}</p>
+                      </div>
+                        <p className="til_post_content">{post.tilPost}</p>
+                    </div>)
+                )}
+            </>
+       )
+    }
+
+
```
Comment (elstevens23) [inline src/components/profile/profile.js]: at this point im bedridden

File: src/components/profile/profile.js [hunk 9]
```diff
@@ -64,15 +80,125 @@ class Profile extends Component {
...
+                this.getTilPosts()
+            } else {
+                this.setState({databaseError: 'Sorry your TIL wasn\'t submitted, try again later.'})
+            }
+        }).catch( err => {
+            this.setState({databaseError: 'Sorry your TIL wasn\'t submitted, try again later.'})
+        })
+    }
+
+
```
Comment (elstevens23) [inline src/components/profile/profile.js]: *dead*

File: src/components/timeline/timeline.js [hunk 10]
```diff
@@ -1,11 +1,11 @@
 import React, { Component } from 'react';
 import '../../App.css';
-import {Link, Redirect} from "react-router-dom";
+import {Redirect} from "react-router-dom";
 
 export default class Timeline extends Component{
-    constructor(props) {
-        super(props);
-    }
+    // constructor(props) {
```
Comment (elstevens23) [inline src/components/timeline/timeline.js:6]: is this necessary

File: src/components/timeline/timeline.js [hunk 11]
```diff
@@ -1,11 +1,11 @@
...
+    //     super(props);
+    // }
 
     //This function makes requests runs after the component loads which will be usefull to load timelines info.
     // componentDidMount() {
```
Comment (elstevens23) [inline src/components/timeline/timeline.js:10]: tf

File: src/components/profile/profile.js (hunk 7 above)
Comment (jennymegan) [reply src/components/profile/profile.js]: ha. ha. ha

File: src/components/timeline/timeline.js (hunk 11 above)
Comment (elstevens23) [reply src/components/timeline/timeline.js:10]: also on lower lines, we need to change isAuthenticated as it is using localStorage

File: src/components/timeline/timeline.js (hunk 10 above)
Comment (jennymegan) [reply src/components/timeline/timeline.js:6]: yes will need it in about ten mins

File: src/components/timeline/timeline.js (hunk 11 above)
Comment (jennymegan) [reply src/components/timeline/timeline.js:10]: again will need in about ten mins

File: src/components/userInfo/index.js [hunk 12]
```diff
@@ -1,17 +1,16 @@
...
 import {Paragraph} from "../paragraph";
-import axios from "axios";
 import './index.css';
 
 export class UserInfo extends Component {
 
-
     render() {
         return (
-            <div className="userInfo">
```
Comment (elstevens23) [inline src/components/userInfo/index.js:12]: as you've removed this class name surely the CSS won't work

**Review by elstevens23 — CHANGES_REQUESTED** [review]
sorry in advance for all my passive agressive comments its smellie's fault

File: src/components/timeline/timeline.js (hunk 11 above)
Comment (elstevens23) [reply src/components/timeline/timeline.js:10]: wha?

File: src/components/userInfo/index.js (hunk 12 above)
Comment (jennymegan) [reply src/components/userInfo/index.js:12]: have applied the class in the level above instead do not panic.

**Review by elstevens23 — APPROVED** [review]

**Pull request merged**